Visualizations/Visualization.py

```diff
@@ -1,50 +1,136 @@
 import networkx as nx
 import matplotlib.pyplot as plt
 import json
+import sqlite3
+from Algorithms import ShortestPath
 
 
 class Visualizer:
+    def __init__(self):
+        self.db = sqlite3.connect("D:\PyCharm\PyCharm 2023.2.4\JakNieDojade\mpk.db")
+        self.cursor = self.db.cursor()
 
-    def get_nodes(self, filepath):
+    def get_nodes_from_graph(self, graph: list, start=0, end=0) -> tuple:
         """
-        Returns a list with connections (i,j) where i,j represent node numbers with a connection and a dictionary of node sizes where {i:number_of_connections *10}.
+        Converts a graph in adjacency matrix form into a collection of connections between nodes.
+        :param graph: a graph in adjacency matrix form.
+        :param start: node IDs for showing the shortest path between them on the graph.
+        :param end: node IDs for showing the shortest path between them on the graph.
+        :returns: returns (all connections, connections in the shortest path between stops given).
         """
-        file = open(filepath, "r")
-        g = json.load(file)
-        graph = g[0]["graph"]
+        sp = ShortestPath.ShortestPath()
         connections = []
-        node_size = {}
+        nodes_in_path = []
 
-        for i in range(len(graph)):
-            counter = 0
-            for j in range(0, len(graph)):
-                if i != j:
-                    if graph[i][j] != 0:
+        if start != 0 and end != 0:
+            path = sp.dijkstra(graph, start, end)[0]
+
+            for i in range(len(path) - 1):
+                nodes_in_path.append((path[i], path[i + 1]))  # add adjacent nodes from the path extracted by the algorithm to the graphs nodes
+
+        else:
+            path = []
+
+        for j in range(len(graph)):
+            for k in range(len(graph)):
+                if j != k:
+                    if graph[j][k] != 0:
+                        connections.append((j, k))  # append connections between two nodes represented by j,k
+        return connections, nodes_in_path
+
+    def get_node_sizes(self, graph: list) -> dict:
+        """
+        Adjusts node sizes in a graph based on the number of possible destinations from the stop that the node represents.
+        :param graph: a graph in adjacency matrix form.
+        :returns: returns a dictionary of node sizes corresponding to every node index.
+        """
+        node_sizes = {}
+        for j in range(len(graph)):
+            counter = 0  # count the amount of connections from every node
+            for k in range(len(graph)):
+                if j != k:
```

Comment:
tak stricte wizualnie pls daj "if j != k and graph[j][k] != 0:" skoro nie ma elsa xD